openresty-build-tools/kong-ngx-build

```diff
@@ -158,44 +168,74 @@ main() {
 
   NGINX_CORE_VER=$(parse_nginx_core_version $OPENRESTY_VER)
 
-  OPENSSL_DOWNLOAD=$DOWNLOAD_CACHE/openssl-$OPENSSL_VER
-  OPENRESTY_DOWNLOAD=$DOWNLOAD_CACHE/openresty-$OPENRESTY_VER
-
   mkdir -p $DOWNLOAD_CACHE $PREFIX
-
-  OPENSSL_DESTDIR=${OPENSSL_DESTDIR:-/}
+  OPENRESTY_DOWNLOAD=$DOWNLOAD_CACHE/openresty-$OPENRESTY_VER
   OPENRESTY_DESTDIR=${OPENRESTY_DESTDIR:-/}
-
-  OPENSSL_PREFIX=${OPENSSL_PREFIX:-$PREFIX/openssl}
   OPENRESTY_PREFIX=${OPENRESTY_PREFIX:-$PREFIX/openresty}
-
-  OPENSSL_INSTALL=$(canon_path $OPENSSL_DESTDIR/$OPENSSL_PREFIX)
   OPENRESTY_INSTALL=$(canon_path $OPENRESTY_DESTDIR/$OPENRESTY_PREFIX)
 
-  if version_lt $NGINX_CORE_VER 1.13.6 || version_lt $OPENSSL_VER 1.1; then
+  if [ -n "$OPENSSL_VER" ]; then
+    OPENSSL_DOWNLOAD=$DOWNLOAD_CACHE/openssl-$OPENSSL_VER
+    OPENSSL_DESTDIR=${OPENSSL_DESTDIR:-/}
+    OPENSSL_PREFIX=${OPENSSL_PREFIX:-$PREFIX/openssl}
+    OPENSSL_INSTALL=$(canon_path $OPENSSL_DESTDIR/$OPENSSL_PREFIX)
+
+    if version_lt $OPENSSL_VER 1.1; then
+      # unconditionally disable module in older core since they are never tested
+      KONG_NGINX_MODULE=0
+    fi
+  fi
+
+  if [ -n "$BORINGSSL_VER" ]; then
+    OPENSSL_DOWNLOAD=$DOWNLOAD_CACHE/boringssl-$BORINGSSL_VER
+#    OPENSSL_DESTDIR=${OPENSSL_DESTDIR:-/}
+#    OPENSSL_PREFIX=${OPENSSL_PREFIX:-$PREFIX/openssl}
+#    OPENSSL_INSTALL=$(canon_path $OPENSSL_DESTDIR/$OPENSSL_PREFIX)
+  fi
+
+  if version_lt $NGINX_CORE_VER 1.13.6; then
```

Comment:
The previous version of this if had an extra condition:

```
if version_lt $NGINX_CORE_VER 1.13.6 || version_lt $OPENSSL_VER 1.1; then
```

Is the `version_lt $OPENSSL_VER 1.1` part not needed any more? I expected it to "reappear" on the `if [ -n "$OPENSSL_VER" ];` conditional, below.

Reply:
boringssl is forked from 1.0.something.  i'll move this condition to the openssl part

Reply:
the full version of this condition is a few lines above